web/libs/core/src/lib/Tour/TourProvider.tsx

```diff
@@ -0,0 +1,181 @@
+  key: new Date().toISOString(),
+  run: false,
+  continuous: true,
+  loading: false,
+  stepIndex: 0,
+  steps: steps.map((step) => ({
+    ...step,
+    // TODO: although html is predefined by LSE assets, to avoid XSS, we should better sanitize it by allowing only simple html tags like <b>, <i>, <u> etc.
+    title: typeof step.title === "string" ? <div dangerouslySetInnerHTML={{ __html: step.title }} /> : step.title,
+    content:
+      typeof step.content === "string" ? <div dangerouslySetInnerHTML={{ __html: step.content }} /> : step.content,
+  })),
+});
+
+type ProductTourState = "ready" | "skipped" | "completed";
+
+const updateProductTourState = (
+  api: any,
+  name: string,
+  state: ProductTourState,
+  interactionData: Record<string, any> = {},
+) => {
+  return api.callApi("updateProductTour", {
+    params: {
+      name,
+    },
+    body: {
+      state,
+      interaction_data: interactionData,
+    },
+  });
+};
+
+interface TourContextType {
+  registerTour: (name: string, dispatch: React.Dispatch<TourAction>) => void;
+  unregisterTour: (name: string) => void;
+  startTour: (name: string) => void;
+  setTourViewed: (name: string, isSkipped: boolean, interactionData: Record<string, any>) => void;
+  restartTour: (name: string) => void;
+}
+
+export const TourContext = createContext<TourContextType | null>(null);
+
+export const TourProvider: React.FC<{
+  children: React.ReactNode;
+  useAPI: () => { callApi: (apiName: string, params: Record<string, any>) => any };
+}> = ({ children, useAPI }) => {
+  const api = useAPI();
+  const toursRef = useRef<Record<string, React.Dispatch<TourAction>>>({});
+
+  const registerTour = (name: string, dispatch: React.Dispatch<TourAction>) => {
+    toursRef.current[name] = dispatch;
+  };
+
+  const startTour = useCallback(
+    async (name: string) => {
+      const dispatch = toursRef.current[name];
+      if (!dispatch) {
+        console.error("Dispatch for tour", name, "not found");
+        return;
+      }
+
+      const response = await api.callApi("getProductTour", { params: { name } });
+
+      if (response?.$meta?.status !== 200) {
+        console.error("Error fetching tour data", response);
+        return;
+      }
+
+      if (!response.steps?.length) {
+        console.info(`No steps found for tour "${name}"`);
+        return;
+      }
+
+      if (response.state === "completed" || response.state === "skipped") {
+        console.info(`Tour "${name}" is already completed`);
+        return;
+      }
+
+      const state = createInitialState(response.steps);
+      dispatch({ type: "GOTO", payload: { ...state, run: true } });
+    },
+    [api],
+  );
+
+  const setTourViewed = useCallback(
+    (name: string, isSkipped: boolean, interactionData: Record<string, any> = {}) => {
+      // TODO: currently we don't have per-tour complete state, so we just update the global state
+      updateProductTourState(api, name, isSkipped ? "skipped" : "completed", interactionData);
```

Comment:
`updateProductTourState` returns a promise that we're not awaiting here or in `restartTour` which means we fire the request and dont care if it resolves as expected or not - is that intended?

Reply:
I think so, it depends on the particular usage of this function. Right now, we just need to set flag on backend to prevent tour reappearing next time. But this can be done async, no reason for waiting.  But I'll put it inside updateProductTourState for clarity